include/psen_scan_v2/diagnostics.h

```diff
@@ -0,0 +1,173 @@
+  POWER_SUPPLY,
+  NETW_PRB,
+  DUST_CRC_FL,
+  OSSD2_OVERCUR,
+  MEAS_PROB,
+  INCOHERENCE,
+  ZONE_INVAL_TRANS,
+  ZONE_INVALID_CONF,
+  WIN_CLN_WARN,
+  INT_COM_PRB,
+  GENERIC_ERR,
+  DISP_COM_PRB,
+  TEMP_MEAS_PROB,
+  ENCOD_OOR,
+  EDM2_ERR,
+  EDM1_ERR,
+  CONF_ERR,
+  OUT_OF_RANGE_ERR,
+  TEMP_RANGE_ERR,
+  ENCODER_GENERIC_ERR,
+  _,
+  UNUSED
+};
+
+class MonitoringFrameDiagnosticMessage
+{
+public:
+  using ErrorBitLocation = uint8_t;
+  using ErrorByteLocation = uint16_t;
+
+  MonitoringFrameDiagnosticMessage(ScannerId id, ErrorByteLocation byte_location, ErrorBitLocation bit_location);
+
+  bool operator==(const MonitoringFrameDiagnosticMessage& rhs) const;
+
+  friend std::array<uint8_t, DIAGNOSTIC_DATA_LENGTH_IN_BYTES>
+  serializeDiagnosticMessages(std::vector<MonitoringFrameDiagnosticMessage>& messages);
+
+  ScannerId getScannerId() const
+  {
+    return id_;
+  }
+
+  DiagnosticCode getDiagnosticCode() const
+  {
+    return code_;
+  }
+
+  ErrorByteLocation getErrorByteLocation() const
+  {
+    return byte_location_;
+  }
+
+  ErrorBitLocation getErrorBitLocation() const
+  {
+    return bit_location_;
+  }
+
+private:
+  ScannerId id_;
+  DiagnosticCode code_;
+  ErrorByteLocation byte_location_;
+  ErrorBitLocation bit_location_;
+};
+
+// clang-format off
+
+typedef DiagnosticCode Dc;
+using ErrorCodeName = std::string;
+
+static const std::map<DiagnosticCode, ErrorCodeName> error_code_to_string
+{
+  { Dc::OSSD1_OC, "OSSD1 Overcurrent / Short circuit." },
+  { Dc::OSSD_SHRT_C, "Short circuit between at least two OSSDs." },
+  { Dc::OSSD_INTEGR, "Integrity check problem on any OSSD" },
+  { Dc::INT, "Internal error." },
+  { Dc::WIN_CLN_AL, "Alarm: The front panel of the safety laser scanner must be cleaned." },
+  { Dc::POWER_SUPPLY, "Power supply problem." },
+  { Dc::NETW_PRB, "Network problem." },
+  { Dc::DUST_CRC_FL, "Dust circuit failure" },
+  { Dc::OSSD2_OVERCUR, "OSSD2 Overcurrent / Short circuit." },
+  { Dc::MEAS_PROB, "Measurement Problem." },
+  { Dc::INCOHERENCE, "Incoherence Error" },
+  { Dc::ZONE_INVAL_TRANS, "Zone: Invalid input - transition or integrity." },
+  { Dc::ZONE_INVALID_CONF, "Zone: Invalid input configuration / connection." },
+  { Dc::WIN_CLN_WARN, "Warning: The front panel of the safety laser scanner must be cleaned." },
+  { Dc::INT_COM_PRB, "Internal communication problem." },
+  { Dc::GENERIC_ERR, "Generic Error." },
+  { Dc::DISP_COM_PRB, "Display communication problem." },
+  { Dc::TEMP_MEAS_PROB, "Temperature measurement problem." },
+  { Dc::ENCOD_OOR, "Encoder: Out of range." },
+  { Dc::EDM2_ERR, "Error in the External Device Monitoring (EDM2_ERR)." },
+  { Dc::EDM1_ERR, "Error in the External Device Monitoring (EDM1_ERR)." },
+  { Dc::CONF_ERR, "Configuration Error." },
+  { Dc::OUT_OF_RANGE_ERR, "Out of range error." },
+  { Dc::TEMP_RANGE_ERR, "Temperature out of range." },
+  { Dc::ENCODER_GENERIC_ERR, "Encoder: Generic error." },
+  { Dc::_, "Unexpected error" } \
```

Comment:
@jschleicher works for you?

Reply:
Yep.
Tiny question: What's the difference between `Dc::_` and `Dc::UNUSED`

Reply:
Not sure. The last 3 bytes are described as "unused" but in the first 6 bytes some entries are marked "-" not sure if this is the same...

Reply:
I removed `Dc::_`